engine/src/main/java/org/hibernate/validator/internal/engine/path/PathImpl.java

```diff
@@ -113,11 +113,11 @@ public PathImpl getPathWithoutLeafNode() {
 		return new PathImpl( nodeList.subList( 0, nodeList.size() - 1 ) );
 	}
 
-	public NodeImpl addPropertyNode(String nodeName) {
+	public NodeImpl addPropertyNode(String nodeName, String resolvedNodeName) {
```

Comment:
keeping just one name as mentioned in one of the above comments would allow us to keep these methods unchanged.